File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int n = s.length();

        vector<int>freq1(26);
        vector<int>freq2(26);

        for(char ch : s){
            freq1[ch-'a']++;
        }

        for(char ch : t){
            freq2[ch-'a']++;
        }

        if(freq1==freq2){
            return true;
        }

        return false;
    }
};

File: 242-valid-anagram/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        
        if(len(s)!=len(t)):
            return False

        freq1 = {}
        freq2 = {}

        for s1 in s:
            freq1[s1] = freq1.get(s1,0) + 1
        
        for s2 in t:
            freq2[s2] = freq2.get(s2,0) + 1
        
        if(freq1 == freq2):
             return True

        return False